charts/refinery/templates/deployment.yaml

```diff
@@ -50,6 +50,7 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           command:
             - "refinery"
+            - "-d"
```

Comment:
Is there a performance cost to enabling this by default?  Instead of enabling this by default I think we could define a new root-level, debug-specific field that when enabled would set this flag and the necessary port.

Reply:
No cost to doing it (unless something is pulling pprof continuously and even then <1% cpu impact); it's purely a security thing, but if people can poke arbitrary ports on your k8s cluster's pods without using the ingress you have other things severely wrong.

Reply:
Maybe it is because of how we've named to capability, but it does not feel proper to turn on debug mode by default.  This mode enables profiling, but also enables some more verbose logging, at least during startup.

I think I'd still prefer a new `debug` section in the values.yaml to configure this value and the port.   Something like

```yaml
debug:
  enabled: false
  port: 6060
```

We can introduce some logic to set `DebugServiceAddr` in the config if `debug.port` is set and `DebugServiceAddr` is not set and/or fail the install via NOTES.txt if  `debug.port` does not equal `DebugServiceAddr`.

Reply:
I don't understand Helm enough to do this, but if you want to modify my PR to do this feel free :)

Reply:
With Refinery 2.0 release, Q3 planning, and hack week I don't anticipate being able to implement the changes myself this week. What is the priority of this work?

Reply:
System optimization work I was hoping to have for https://github.com/honeycombio/infra/pull/4432 - which I'm hoping to do before I go on sabbatical this week; if we can for now run our internal refineries on k8s off this branch for this week before landing, then no rush at all.

Reply:
@RainofTerra can we do that?

Reply:
@lizthegrey if that is possible you should still update your branch to explicitly set DebugServiceAddr even if long term  the api for this feature in the helm chart changes.

Reply:
We'd need to publish a --devel version of the chart to the https://honeycombio.github.io/helm-charts/ repo. I can try to investigate.

Reply:
Pushed as v1.9.2-hnyinternal.0

Reply:
> @lizthegrey if that is possible you should still update your branch to explicitly set DebugServiceAddr even if long term the api for this feature in the helm chart changes.

Done in https://github.com/honeycombio/infra/pull/4432

Reply:
Added the debug section.